### ISSUE TITLE
Files being cached and new changes not compiled

### ISSUE THREAD

**Issue by k-may** [ivogabe/gulp-typescript#373]
Very strange, seems that my .ts files are being cached somewhere. While using watch, if I delete the compiled files, change the source .ts and look at the newly compiled files, my changes aren't reflected. In fact the compiled files still reflect the same out dated code and only compiles as expected when I re-run the task.  Somehow, my old files must be being cached somewhere? 

**gulpfile**

```javascript

var tsProject = ts.createProject('tsconfig.json', {
    declarationFiles: true
});

gulp.task('ts', function () {

    gulp.watch("src/**/*.ts", function () {
        var tsResult = tsProject.src()
            .pipe(sourcemaps.init())
            .pipe(ts(tsProject));


        merge(tsResult.js
                .pipe(sourcemaps.write('.'))
                .pipe(gulp.dest('.'))
                .pipe(debug({title: "compile"})),
            tsResult.dts
                .pipe(gulp.dest('.'))
                .pipe(debug({title: "compile"}))
        )

    });

});
```

**tsconfig.json**

```json
{
  "compilerOptions": {
    "module": "system",
    "target": "es5",
    "sourceMap": true,
    "outFile": "./bin/OrderForm.js",
    "watch": true,
    "diagnostics": true,
    "pretty": true,
    "listFiles": true,
    "rootDir": "OrderForm/"
  },
  "exclude": [
    "node_modules",
    "bin",
    "lib"
  ],
  "files" :[
    "src/OrderForm.ts"
  ]
}
```

**Comment by ivogabe** [comment]
I've tried your gulpfile, but it works fine here. Can you create a small example, zip it and upload it here? And can you print the output of the console here?

**Comment by k-may** [comment]
I've created a small project, but currently am unable to reproduce it there, but I'm still seeing behavior in my actual project. Very strange. If I delete the compiled files, make a change in the source code, look at the generated files they just aren't reflecting the changes. I only see the changed if I stop and start the gulp task and make an edit to the source?? If I can find a way to reproduce it for you I'll post here.

**Comment by nthansen** [comment]
I was having the same issue and I realized a few weeks ago I set my VS Code settings to ignore my source .js files where there's a .ts file name that matches. I ran the debugger with iron node and realized it was copying over the .js file instead. 

Once I deleted the javascript files in my source folder, everything is working fine again. 

I'm using commonjs for my module and "allowJs": true. Hope that helps.

**Comment by ivogabe** [comment]
I think that this issue does not exist in gulp-typescript 3.0. Can everyone here verify that?

**Comment by josdejong** [comment]
We had the same issue. I can confirm that this issue is fixed in `gulp-typescript 3.0.1`.

**Comment by ivogabe** [comment]
Great!
